src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs

```diff
@@ -70,7 +70,7 @@ public void Dispose()
             if (solutionState.TryGetStateChecksums(out var stateChecksums))
                 await stateChecksums.FindAsync(solutionState, assetHint, remainingChecksumsToFind, result, cancellationToken).ConfigureAwait(false);
 
-            foreach (var projectId in solutionState.ProjectIds)
+            foreach (var projectId in solutionState.Solution.ProjectIds)
```

Comment:
The naming here all seems odd -- solutionState is now a SolutionCompilationState, neither of which actually really have a pointer to the Solution...